jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/HttpChannel.java

```diff
@@ -463,15 +462,11 @@ public void clearAttributes()
             _requestAttributes.clearAttributes();
         }
 
-        @Override
-        public void execute(Runnable task)
-        {
-            _server.getThreadPool().execute(task);
-        }
-
         @Override
         public String getId()
         {
+            if (_id == null)
+                _id = getConnectionMetaData().getId() + "#" + _stream.getId();
```

Comment:
Not sure about this now. `_stream` may be `null` and must be read with a lock held, so maybe not worth it to use the `_stream` to generate the id of a request.

Reply:
The whole ID thing needs thought and review.  It was never the subject of this PR.  Can't we punt that until we actually pin down what the ID requirements are?